warzone/MapLoader.cpp

```diff
@@ -306,4 +304,332 @@ void MapLoader::createBorder(std::string border, Map* map)
+	{
+		return true;
+	}
+	else {
+		return false;
+	}
+}
+
+//Check if the format of the territory is fine
+bool ConquestFileReader::checkTerritory(std::string line)
+{
+	std::stringstream ss(line);
+	std::vector<std::string> result;
+
+	while (ss.good())
+	{
+		std::string substr;
+		getline(ss, substr, ',');
+		result.push_back(substr);
+	}
+	ss.clear();
+
+	if (!isdigit(result.at(1).back()) || !isdigit(result.at(2).back())) {
+
+		return false;
+	}
+
+	if (result.size() >= 5 && result.size() <= 14) {
+		return true;
+	}
+	else {
+		return false;
+	}
+}
+
+Continent* ConquestFileReader::createContinent(std::string continent)
+{
+	//Declare variables that will be used to store the calues from ss
+	std::string name;
+	std::string colour = "White";
+	unsigned int value = 0;
+
+	
+	name = continent.substr(0, continent.find('='));
+	value = continent.back() - '0'; //NOICE
+	Continent* myContinent = new Continent(name, colour, value);
+	return myContinent;
+}
+
+Territory* ConquestFileReader::createTerritory(std::string territory, Map* map)
+{
+	// Declare variables that will be used to store the calues from ss
+	std::string name;
+	Continent* continent;
+	unsigned int x;
+	unsigned int y;
+
+	std::stringstream ss(territory);
+	std::vector<std::string> result;
+
+	while (ss.good())
+	{
+		std::string substr;
+		getline(ss, substr, ',');
+		result.push_back(substr);
+	}
+	ss.clear();
+
+	//Put each input into the right variable. The syntax is always the same, thus we can do that
+	name = result.at(0);
+	x = std::stoul(result.at(1));
+	y = std::stoul(result.at(2));
+	continent = map->getContinent(result.at(3));
+
+	Territory* myTerritory = new Territory(name, continent, x, y);
+
+	return myTerritory;
+}
+
+void ConquestFileReader::createBorder(std::string border, Map* map){
+	// Declare variables that will be used to store the calues from ss
+	std::string name;
+	std::vector<Territory*> neighbors;
+	Territory* territory;
+
+	std::stringstream ss(border);
+	std::vector<std::string> result;
+
+	while (ss.good())
+	{
+		std::string substr;
+		getline(ss, substr, ',');
+		result.push_back(substr);
+	}
+	ss.clear();
+
+	name = result.at(0);
+
+	//Find the pointer to territory
+	territory = map->getTerritory(name);
+
+	for (int i = 4; i < result.size(); i++) {
+		neighbors.push_back(map->getTerritory(result.at(i)));
+	}
+
+	int index = map->getTerritoryID(name);
+
+	//Add the border to the map
+	map->addBorder(index, &neighbors);
+}
+
+// Copy constructor.
+ConquestFileReaderAdapter::ConquestFileReaderAdapter(ConquestFileReaderAdapter* adapter)
+{
+	this->fileName = std::string(adapter->fileName);
+	this->conquestMapLoader = new ConquestFileReader("");
+}
+
+// Param constructor.
+ConquestFileReaderAdapter::ConquestFileReaderAdapter(ConquestFileReader* cfr) : MapLoader(fileName), conquestMapLoader(cfr) { }
+
+// Destructor.
+ConquestFileReaderAdapter::~ConquestFileReaderAdapter()
+{
+	if (!conquestMapLoader) {
+		delete conquestMapLoader;
+	}
```

Comment:
I feel like this check might lose us points.
You can remove it and just `delete conquestMapLoader` if you stop deleting `conqLoader` at `GameEngine.cpp` line 92. I just checked and it seems to work.

Reply:
Yeah I didn't think it was good and I know that would work but idk if they would be unhappy about a `new` without a `delete`

Reply:
But I think this makes sense.